clients/fides-js/src/docs/fides-event.ts

```diff
@@ -104,6 +104,11 @@ export interface FidesEvent extends CustomEvent {
        */
       servingComponent?: "banner" | "modal" | "tcf_banner" | "tcf_overlay";
 
+      /**
+       * Whether the user should be shown the consent experience. Only available on FidesInitialized events.
```

Comment:
nit: Any reason why we can't include this on all events? It'd be easier to understand...

Reply:
Well, I think this is only really useful for the initialized event and to get this on the other events, I'd have to pass FidesGlobal down to all of the code that emits them. So there's a reason - I didn't say it was a good one :)

Reply:
Heh, I'm not buying it! The type signature for dispatchFidesEvent should just require the FidesGlobal as an arg then, I think it has only like 5 unique call sites? I suspect all of those are already just plucking from the FidesGlobal to provide args to that function today, so the code is telling us it should be zhushed slightly...

Reply:
I'm in agreement that this is only really useful for the initialized event. Correct me if I'm wrong @NevilleS, but all other events happen as a result of the user being shown the experience already so it's too late to be useful.

Reply:
Here's my final defense:
1. API Consistency: It's _easier_ to understand `FidesEvent` if the properties are the same across all events. As a developer, I don't have to learn about the "special cases", I can just look at `detail.extraDetails.shouldShowExperience` and trust that, whether my code is subscribed to `FidesUpdated` or `FidesInitialized` or `FidesUpdating`, etc.
2. Utility: It's also useful to have `shouldShowExperience: false` be triggered by `FidesUpdated` once someone's consent preferences are saved. If you're building your own show/hide logic for a custom banner, you can use that as a state variable... 👍

I can't *guarantee* that we need it though, so we could ship this as-is and add support to other events later if it becomes important; but we'd also need to have a good explanation for when to "hide" a custom experience after it's been shown...

Reply:
Anyways I should be clear!

We can merge this as-is (well, once CI checks are passing). But I think developer using this API for their own apps will find it a bit confusing. Happy to be wrong there though 👍

Reply:
I gave this a solid attempt and it started to turn into a much larger refactor. Basically, everywhere we pass down options, cookie, experience and saved_consent, which is throughout the UI components, the FidesGlobal can be used instead. Some of those code paths emit events while others don't, but it would be strange to keep using different patterns for the same state variables.